File: backend/TripService/Models/Activity.cs
using System;

namespace TripService.Models
{
    public class Activity
    {
        public Guid Id { get; set; }

        public Guid TripPlanId { get; set; }

        public TripPlan TripPlan { get; set; }

        public string Name { get; set; }

        /// <summary>Datum i vreme aktivnosti (kombinovano u jednom polju).</summary>
        public DateTime DateTime { get; set; }

        public string Location { get; set; }

        public string Description { get; set; }

        public decimal EstimatedCost { get; set; }

        public ActivityStatus Status { get; set; }

        public DateTime CreatedAt { get; set; }
    }
}

File: frontend/src/models/Activity.ts
export type ActivityStatus = "Planned" | "Reserved" | "Completed" | "Cancelled";

export interface Activity {
  id: string;
  tripPlanId: string;
  name: string;
  dateTime: string;
  location: string;
  description: string;
  estimatedCost: number;
  status: ActivityStatus;
  createdAt: string;
}

export interface ActivityRequest {
  name: string;
  dateTime: string;
  location: string;
  description: string;
  estimatedCost: number;
  status: ActivityStatus;
}

export const ACTIVITY_STATUS_LABELS: Record<ActivityStatus, string> = {
  Planned: "Planirano",
  Reserved: "Rezervisano",
  Completed: "Završeno",
  Cancelled: "Otkazano",
};
